src/renderer/components/dock/logs/__test__/download-logs.test.tsx

```diff
@@ -0,0 +1,186 @@
+    getPodsByOwnerId: jest.fn(),
+    areLogsPresent: jest.fn(),
+    searchStore: new SearchStore(),
+    ...deps,
+  });
+}
+
+const getOnePodViewModel = (tabId: TabId, deps: Partial<LogTabViewModelDependencies> = {}): LogTabViewModel => {
+  const selectedPod = dockerPod;
+
+  const logs = new Map([["timestamp", "some-logs"]]);
+
+  return mockLogTabViewModel(tabId, {
+    getLogTabData: () => ({
+      selectedPodId: selectedPod.getId(),
+      selectedContainer: selectedPod.getContainers()[0].name,
+      namespace: "sadsadaasdad",
+      showPrevious: false,
+      showTimestamps: false,
+    }),
+    getPodById: (id) => {
+      if (id === selectedPod.getId()) {
+        return selectedPod;
+      }
+
+      return undefined;
+    },
+    getTimestampSplitLogs: () => [...logs],
+    getLogs: () => ["some-logs"],
+    getLogsWithoutTimestamps: () => ["some-logs"],
+    ...deps,
+  });
+};
+
+describe("download logs options in pod logs dock tab", () => {
+  let builder: ApplicationBuilder;
+  let openSaveFileDialogMock: jest.MockedFunction<() => void>;
+
+  beforeEach(() => {
+    builder = getApplicationBuilder();
+
+    builder.setEnvironmentToClusterFrame();
+
+    builder.beforeApplicationStart(({ rendererDi }) => {
+      rendererDi.override(callForAllLogsInjectable, () => () => Promise.resolve("all-logs"));
+      rendererDi.override(logsViewModelInjectable, () => getOnePodViewModel("foobar"));
+      rendererDi.override(getRandomIdForPodLogsTabInjectable, () =>
+        jest
+          .fn(() => "some-irrelevant-random-id")
+          .mockReturnValueOnce("some-first-tab-id")
+          .mockReturnValueOnce("some-second-tab-id"),
+      );
+
+      openSaveFileDialogMock = jest.fn();
+      rendererDi.override(openSaveFileDialogInjectable, () => openSaveFileDialogMock);
+    });
+
+  });
+
+  describe("when opening pod logs", () => {
+    let rendered: RenderResult;
+    let rendererDi: DiContainer;
+    
+    beforeEach(async () => {
+      rendered = await builder.render();
+      rendererDi = builder.dis.rendererDi;
+
+      const pod = dockerPod;
+      const createLogsTab = rendererDi.inject(createPodLogsTabInjectable);
+      const container = {
+        name: "docker-exporter",
+        image: "docker.io/prom/node-exporter:v1.0.0-rc.0",
+        imagePullPolicy: "pull",
+      };
+
+      const dockStore = rendererDi.inject(dockStoreInjectable);
+
+      dockStore.closeTab("terminal");
+
+      createLogsTab({
+        selectedPod: pod,
+        selectedContainer: container,
+      });
+    });
+
+    it("renders", () => {
+      expect(rendered.baseElement).toMatchSnapshot();
+    });
+
+    it("contains download dropdown button", () =>  {
+      expect(rendered.getByTestId("download-logs-dropdown")).toBeInTheDocument();
+    });
+
+    describe("when clicking on button", () => {
+      beforeEach(() => {
+        const button = rendered.getByTestId("download-logs-dropdown");
+
+        act(() => button.click());
```

Comment:
Is this act call required?

Reply:
Yes, it removes error notification (https://reactjs.org/docs/test-utils.html#act):
```
Warning: An update to DownloadLogsDropdown inside a test was not wrapped in act(...).
When testing, code that causes React state updates should be wrapped into act(...):
    act(() => {
      /* fire events that update state */
    });
    /* assert on the output */
This ensures that you're testing the behavior the user would see in the browser.
```